server.go

```diff
@@ -1963,38 +1965,57 @@ func SendHeader(ctx context.Context, md metadata.MD) error {
 
 // SetSendCompressor sets a compressor for outbound messages.
 // It must not be called after any event that causes headers to be sent
-// (see SetHeader for a complete list). Provided compressor is used when below
+// (see _ServerStream_.SetHeader for a complete list). Provided compressor is used when below
 // conditions are met:
 //
 //   - compressor is registered via encoding.RegisterCompressor
 //   - compressor name exists in the client advertised compressor names sent in
-//     :grpc-accept-encoding header.
+//     grpc-accept-encoding header. Use _ServerStream_.ClientAdvertisedCompressors
+//     to get client advertised compressor names.
 //
 // The context provided must be the context passed to the server's handler.
+// It must be noted that compressor name "identity" disables the outbound compression.
+// By default, server messages will be sent using the same compressor with which
+// request messages were sent.
 //
-// The error returned is compatible with the status package.  However, the
-// status code will often not match the RPC status as seen by the client
-// application, and therefore, should not be relied upon for this purpose.
 // It is not safe to call SetSendCompressor concurrently with SendHeader and
 // SendMsg.
 //
 // # Experimental
 //
-// Notice: This type is EXPERIMENTAL and may be changed or removed in a
+// Notice: This _function_ is EXPERIMENTAL and may be changed or removed in a
 // later release.
 func SetSendCompressor(ctx context.Context, name string) error {
 	stream, ok := ServerTransportStreamFromContext(ctx).(*transport.Stream)
 	if !ok || stream == nil {
-		return status.Errorf(codes.Internal, "grpc: failed to fetch the stream from the context %v", ctx)
+		return fmt.Errorf("failed to fetch the stream from the given context")
 	}
 
 	if err := validateSendCompressor(name, stream.ClientAdvertisedCompressors()); err != nil {
-		return status.Errorf(codes.Internal, "grpc: unable to set send compressor: %v", err)
+		return fmt.Errorf("unable to set send compressor: %w", err)
 	}
 
 	return stream.SetSendCompress(name)
 }
 
+// ClientAdvertisedCompressors returns compressor names advertised by the client
+// via grpc-accept-encoding header.
+//
+// The context provided must be the context passed to the server's handler.
+//
+// # Experimental
+//
+// Notice: This _function_ is EXPERIMENTAL and may be changed or removed in a
+// later release.
+func ClientAdvertisedCompressors(ctx context.Context) ([]string, error) {
```

Comment:
Nit/bikeshedding: `ClientSupportedCompressors`?  (@easwars - thoughts / other ideas?)

I didn't mention anything before because it was all internal-only, but now that it's an exported API I think we should be careful with naming and brainstorm and try to find the best option.

Reply:
`ClientSupportedCompressors` feels better than.

Reply:
`ClientSupportedCompressors` sounds good. 

I feel we should keep the internal method name on stream as `ClientAdvertisedCompressors` as it holds the raw comma-separated string of compressor names whereas `ClientSupportedCompressors` returns a slice of string.